src/js/controller.js

```diff
@@ -356,67 +356,54 @@ export const MediaUIStates = {
     mediaEvents: ['enterpictureinpicture', 'leavepictureinpicture'],
   },
   MEDIA_IS_CASTING: {
-    // Note this relies on a customized video[is=castable-video] element.
-    get: function (controller, e) {
+    // Note this relies on a customized castable-video element.
+    get: function (controller) {
       const { media } = controller;
 
-      if (!media) return false;
+      if (!media?.remote || media.remote?.state === 'disconnected') return false;
 
-      const castElement = globalThis.CastableVideoElement?.castElement;
-      let castState = containsComposedNode(media, castElement);
-
-      // While the cast is connecting set media-is-cast="connecting"
-      if (e?.type === 'castchange' && e?.detail === 'CONNECTING') {
-        castState = 'connecting';
-      }
-
-      return castState;
+      return media.remote.state;
     },
-    mediaEvents: ['entercast', 'leavecast', 'castchange'],
+    remoteEvents: ['connect', 'connecting', 'disconnect'],
   },
   MEDIA_AIRPLAY_UNAVAILABLE: {
     // NOTE: only adding this if airplay is supported, in part to avoid unnecessary battery consumption per
     // Apple docs recommendations (See: https://developer.apple.com/documentation/webkitjs/adding_an_airplay_button_to_your_safari_media_controls)
-    // For a more advanced solution, we could monitor for media state receivers that "care" about airplay support and add/remove
-    // whenever these are added/removed. (CJP)
-    // NOTE: I don't think only adding this if supported helps, so making this match others.
-    // If Airplay's not supported then it it doesn't hurt to add it.
-    // If battery is really an issue we need a different approach.
-    // Also this is a terrible API, Apple. (heff)
-    get: function (controller, e) {
+    get: function (controller, availability) {
       if (!airplaySupported) return AvailabilityStates.UNSUPPORTED;
```

Comment:
praise: Was originally concerned that we wouldn't be able to disambiguate between chromecast vs. airplay availability states because of the generic nature of the Remote Playback API, but then I realized that these initial supported state gates here and for the chromecast equivalent account for that concern 😎